src/FilePreviewer.js

```diff
@@ -26,206 +51,208 @@ import ViewportContent from './ViewportContent';
+    const newScale = originalSizes.map(originalSize =>
+      getFitToScreenScale(viewportSize, originalSize),
     );
+
+    setFile(prevValue => R.assoc('scale', newScale, prevValue));
+  };
+
+  /**
+   * Handler. Download the current file, it can be a
+   * PDF or Image, as `url` or `base64` content.
+   *
+   * @return {void}
+   */
+  const onDownload = () => {
+    const url = file.url || `data:${file.mimeType};base64,${file.data}`;
+    return saveFile(url, file.name || 'download.pdf');
+  };
+
+  /**
+   * Handler. Callback after the PDF gets
+   * processed successfully by `react-pdf`.
+   *
+   * @param  {Object} pdf
+   * @return {void}
+   */
+  const onLoadSuccess = pdf => {
+    // Wait until the original sizes gets calculated.
+    const promises = R.times(async i => {
+      // Wait until the original sizes gets calculated.
+      const page = await pdf.getPage(i + 1);
+      return R.pick(['width', 'height'], page.getViewport({ scale: 1 }));
+    }, pdf.numPages);
+
+    // Set the original sizes for this file.
+    Promise.all(promises).then(setOriginalSizes);
+
+    setTotalPages(pdf.numPages);
+    setUsingFitToScreen(true);
+  };
+
+  // Set an effect to process the file resizing when the
+  // viewport sizes changes and it's using "fit to screen".
+  useEffect(() => {
+    // Check if it's using "fit to screen".
+    if (usingFitToScreen) {
+      // Get the "fit to screen" scale.
+      const newScale = originalSizes.map(originalSize =>
+        getFitToScreenScale(viewportSize, originalSize),
+      );
+
+      // Update the scale.
+      setFile(prevValue => R.assoc('scale', newScale, prevValue));
+    }
+
+    return () => {};
+  }, [usingFitToScreen, originalSizes, viewportSize]);
+
+  return (
+    <div ref={viewportRef} onClick={props.onClick} className="preview-wrapper">
+      <PreviewBar
+        onPageUp={onPageUp}
+        onRotate={onRotate}
+        totalPages={totalPages}
+        onDownload={onDownload}
+        onPageDown={onPageDown}
+        currentPage={currentPage}
+        hidden={props.hideControls}
+      />
+
+      <ViewportContent
+        file={file}
+        totalPages={totalPages}
+        containerRef={containerRef}
+        onPageChange={setCurrentPage}
+        onLoadSuccess={onLoadSuccess}
+      />
+
+      <ViewportControl
+        onZoomIn={onZoomIn}
+        onZoomOut={onZoomOut}
+        onFitToScreen={onFitToScreen}
+        hidden={props.hideControls}
+      />
+    </div>
+  );
 };
 
 FilePreviewer.propTypes = {
-    file: PropTypes.shape({
-        url: PropTypes.string,
-        mimeType: PropTypes.string,
-        data: PropTypes.string,
-        name: PropTypes.string,
-    }).isRequired,
-    onClick: PropTypes.func,
-    hideControls: PropTypes.bool
+  file: PropTypes.shape({
+    url: PropTypes.string,
+    mimeType: PropTypes.string,
+    data: PropTypes.string,
+    name: PropTypes.string,
+  }).isRequired,
+  onClick: PropTypes.func,
+  totalPages: PropTypes.number,
```

Comment:
When would this ever be passed in by the component implementation? The component should figure this out internally. Also, it doesn't look like it's ever used.

Reply:
Good catch, thanks!

Reply:
Doesn't look like this was taken out.